awscli/utils.py

```diff
@@ -57,6 +61,10 @@ def initialize(self):
     def __getattr__(self, item):
         return getattr(self.initialize(), item)
 
+    def communicate(self, *args, **kwargs):
+        if self._process is not None or args or kwargs:
```

Comment:
So if process is `None`, and `args` and `kwargs` are not provided this does nothing? Is that what we want? There is a lot of code that does this to wait for the process to terminate.

Reply:
Because it's a pager process we spawn it, write to its stdin and communicate at the end. Due to it's lazy nature process will be created only in case we actually wrote something to the stream and if by the moment we get to the communicate step process hasn't been created yet it means nothing was written and we can just omit this step and finish.